### ISSUE TITLE
Make it easy to get the screen pixel from a coordinate

### ISSUE THREAD

**Issue by eric-g-97477** [openlayers/openlayers#16462]
**Is your feature request related to a problem? Please describe.**
I asked the following question on StackOverflow:

[How to get the correct screen pixel from a coordinate with OpenLayers?](https://stackoverflow.com/questions/79275965/how-to-get-the-correct-screen-pixel-from-a-coordinate-with-openlayers)

**Describe the solution you'd like**

I think it would be useful to add a function similar to `getPixelFromCoordinate`, but have it return the pixel in screen coordinates.

The answer to the question has the solution for the EPSG:4326 projection, but providing a generic solution for any arbitrary projection would be useful.

**Comment by eric-g-97477** [comment]
> Why not just
> 
> ```js
> map.on("click", (event) => {
>   if (event.dragging) {
>     return;
>   }
> 
>   const screenPixel = [event.originalEvent.clientX, event.originalEvent.clientY];
>   console.log(screenPixel);
> });
> ```
> 
> Or, if you need that outside an event handler:
> 
> ```js
> const pixel = map.getPixelFromCoordinate(coordinate);
> const mapEl = map.getTargetElement();
> const screenPixel = [pixel[0] + mapEl.clientLeft, pixel[1] + mapEl.clientTop];
> ```

I need it outside of the event handler.

The method you suggest does not work.

**Comment by ahocevar** [comment]
What in particular does not work?

**Comment by eric-g-97477** [comment]
> What in particular does not work?

When you are zoomed and have wrapped around a few times, the pixel[0] will be a very large negative number.

See the referenced question and try it yourself.

**Comment by ahocevar** [comment]
I see. Knowing that [toLonLat()](https://github.com/openlayers/openlayers/issues/9790#issuecomment-513367103) normalizes the world coordinate, a (maybe not very obvious) solution is
```js
const pixel = map.getPixelFromCoordinate(fromLonLat(toLonLat(coordinate)));
const mapEl = map.getTargetElement();
const screenPixel = [pixel[0] + mapEl.clientLeft, pixel[1] + mapEl.clientTop];
```

**Comment by eric-g-97477** [comment]
> I see. Knowing that [toLonLat()](https://github.com/openlayers/openlayers/issues/9790#issuecomment-513367103) normalizes the world coordinate, a (maybe not very obvious) solution is
> 
> ```js
> const pixel = map.getPixelFromCoordinate(fromLonLat(toLonLat(coordinate)));
> const mapEl = map.getTargetElement();
> const screenPixel = [pixel[0] + mapEl.clientLeft, pixel[1] + mapEl.clientTop];
> ```

You can see that idea implemented on my codepen at https://codepen.io/ericg_off/pen/raBMxPJ

Unfortunately, it does not work.

Move the white dot over the USA to the left until it wraps around and appears on the right. Then, click on the map to computed the screenPixel of the coordinate. It will report something like [-83.04551654696246, 38.47152263678535].

The -83 x coordinate is wrong.

**Comment by ahocevar** [comment]
I see. You need the conversion the other way around. We've had similar requests quite a few times. Maybe a good solution would be to add an `options` argument to `map.getPixelFromCoordinate()` and `map.getCoordinateFromPixel()`. Then we could have a `wrapX` option, and you could do the natural thing:
```js
const pixel = map.getPixelFromCoordinate(coordinate, {wrapX: false});
```
Does that sound right?

**Comment by eric-g-97477** [comment]
> I see. You need the conversion the other way around. We've had similar requests quite a few times. Maybe a good solution would be to add an `options` argument to `map.getPixelFromCoordinate()` and `map.getCoordinateFromPixel()`. Then we could have a `wrapX` option, and you could do the natural thing:
> 
> ```js
> const pixel = map.getPixelFromCoordinate(coordinate, {wrapX: false});
> ```
> 
> Does that sound right?

Yes. I think that would be a good solution. I am glad I am not alone in thinking this would be useful.
